checkpoint/side_milestone_handler.go

```diff
@@ -0,0 +1,166 @@
+package checkpoint
+
+import (
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmTypes "github.com/tendermint/tendermint/types"
+
+	"github.com/maticnetwork/heimdall/checkpoint/types"
+	"github.com/maticnetwork/heimdall/common"
+	"github.com/maticnetwork/heimdall/helper"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+)
+
+// SideHandleMsgMilestone handles MsgMilestone message for external call
+func SideHandleMsgMilestone(ctx sdk.Context, k Keeper, msg types.MsgMilestone, contractCaller helper.IContractCaller) (result abci.ResponseDeliverSideTx) {
+	// get params
+	milestoneLength := helper.MilestoneLength
+
+	// logger
+	logger := k.Logger(ctx)
+
+	//Check for the hard fork value
+	if ctx.BlockHeight() < helper.GetMilestoneHardForkHeight() {
+		logger.Error("Network hasn't reached the", "Hard forked height", helper.GetMilestoneHardForkHeight())
+		return common.ErrorSideTx(k.Codespace(), common.CodeInvalidBlockInput)
+	}
+
+	// validate milestone
+	count := k.GetMilestoneCount(ctx)
+	lastMilestone, err := k.GetLastMilestone(ctx)
+
+	if count != uint64(0) && err != nil {
```

Comment:
What if count==0 and err!=nil?

Reply:
On calling the GetLastMilestone(ctx) from keeper,  we get the error only when the milestone doesn't exist in the memory.
If the count is 0 , then we are sure that err is because of non existence on any milestone.

Reply:
What if something will be changed? We should treat calls as black boxes.